src/components/GeneralLookupField.tsx

```diff
@@ -0,0 +1,210 @@
+    if (handleEthAddress) return handleEthAddress(inputString);
+    return {
+      display: inputString,
+      value: inputString
+    };
+  };
+
+  const validateAddress = (v: IReceiverAddress) => {
+    const validationResult = isValidETHRecipientAddress(v.value, resolutionError);
+    if (validationResult.success) return;
+
+    return {
+      name: validationResult.name,
+      type: validationResult.type,
+      message: validationResult.message
+    };
+  };
+
+  useEffectOnce(() => {
+    if (value && value.value && onLoad) {
+      onLoad(setFieldValue);
+    }
+  });
+
+  const handleENSnameDefault = async (
+    inputString: string,
+    defaultInput: IReceiverAddress
+  ): Promise<IReceiverAddress> => {
+    try {
+      const resolvedAddress = await handleDomainResolve(inputString);
+      if (!resolvedAddress) return defaultInput;
+
+      const [label] = inputString.split('.');
+
+      if (handleENSName) return handleENSName(resolvedAddress, inputString);
+
+      return {
+        display: label,
+        value: resolvedAddress
+      };
+    } catch (err) {
+      console.debug(`[GenericLookupField] ${err}`);
+      return { display: '', value: '' };
+    }
+  };
+
+  const handleNewInput = async (inputString: string) => {
+    setResolutionError(undefined);
+    if (!inputString || !network || !network.id || !options) {
+      return;
+    }
+
+    let contact: IReceiverAddress = {
+      display: inputString,
+      value: inputString
+    };
+
+    if (isValidETHAddress(inputString)) {
+      // saves 0x address to address book labeled as "Recipient X"
+      contact = handleEthAddressDefault(inputString);
+    } else if (isValidENSName(inputString)) {
+      // resolves ENS name and saves it to address book labeled as first part of ENS name before "." (example.test.eth --> example)
+      contact = await handleENSnameDefault(inputString, contact);
+    }
+
+    setFieldValue(name, contact, true);
+    setFieldTouched(name, true, false);
+  };
+
+  const handleDomainResolve = async (domain: string): Promise<string | undefined> => {
+    if (!domain || !network) {
+      setIsResolvingDomain(false);
+      setResolutionError(undefined);
+      return;
+    }
+    setIsResolvingDomain(true);
+    setResolutionError(undefined);
+    try {
+      const unstoppableAddress = await UnstoppableResolution.getResolvedAddress(
+        domain,
+        getBaseAssetByNetwork({ network, assets })!.ticker
+      );
+      return unstoppableAddress;
+    } catch (err) {
+      // Force the field value to error so that isValidAddress is triggered!
+      setFieldValue(
+        name,
+        {
+          ...value,
+          value: ''
+        },
+        true
+      );
+      if (UnstoppableResolution.isResolutionError(err)) {
+        setResolutionError(err);
+      } else throw err;
+    } finally {
+      setIsResolvingDomain(false);
+    }
+  };
+
+  const GeneralDropdownFieldCallback = useCallback(() => {
```

Comment:
Can you add some tests for this field to make sure name resolution and inputs are correctly validated and displaying correct error and loading states?